### PR TITLE
ebs br: add concurrency for aws operation & restore schedule ASAP

### LINKED ISSUE #37278: create snapshot/volume concurrently & restore schedule ASAP

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #37278

Problem Summary:

### What is changed and how it works?
- add concurrency for aws operation
- delete snapshot if backup failed
- restore schedule ASAP
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/aws/ebs.go [hunk 1]
```diff
@@ -156,16 +179,23 @@ func (e *EC2Session) DeleteSnapshots(snapIDMap map[string]string) {
...
+				// we don't return error if it fails to make sure all snapshot got chance to delete.
+			}
+			return nil
 		})
-		if err2 != nil {
-			log.Error("failed to delete snapshot", zap.Error(err2))
-			// todo: we can only retry for a few times, might fail still, need to handle error from outside.
-		}
 	}
+	_ = eg.Wait()
```
Comment (3pointer) [inline br/pkg/aws/ebs.go:198]: maybe add some logs here, about how many snapshot deleted?

File: br/pkg/task/restore_ebs_meta.go [hunk 2]
```diff
@@ -43,18 +43,20 @@ const (
...
 type RestoreEBSConfig struct {
 	Config
-	OutputFile       string               `json:"output-file"`
-	SkipAWS          bool                 `json:"skip-aws"`
-	VolumeType       config.EBSVolumeType `json:"volume-type"`
-	VolumeIOPS       int64                `json:"volume-iops"`
-	VolumeThroughput int64                `json:"volume-throughput"`
+	OutputFile          string               `json:"output-file"`
+	SkipAWS             bool                 `json:"skip-aws"`
+	CloudAPIConcurrency uint                 `json:"cloud-api-concurrency"`
```
Comment (3pointer) [inline br/pkg/task/restore_ebs_meta.go:56]: 🤔 shall we expose this config to user?

File: br/pkg/aws/ebs.go (hunk 1 above)
Comment (D3Hunter) [reply br/pkg/aws/ebs.go:201]: https://github.com/pingcap/tidb/pull/37279/commits/93bd3219e9fc4209c13b5079e3e7084d70a5b30a

File: br/pkg/task/restore_ebs_meta.go (hunk 2 above)
Comment (D3Hunter) [reply br/pkg/task/restore_ebs_meta.go:56]: whole command is hidden, see https://github.com/pingcap/tidb/pull/37279/commits/93bd3219e9fc4209c13b5079e3e7084d70a5b30a

Comment (3pointer) [reply br/pkg/task/restore_ebs_meta.go:56]: ok, and I think we need to discuss with @IANTHEREAL if we need to expose any new config to user.

**Review by 3pointer — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
this pr will not compile since there's conflict in kvproto

will merge it for now

**Pull request merged**